File: neta-hiring-poc/dotnet-api/NetaSystems.Hiring.Api/NetaSystems.Hiring.Api/Entities/EmployeeDocument.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NetaSystems.Hiring.Api.Entities
{
    [Table("EmployeeDocuments")]
    public class EmployeeDocument
    {
        [Key]
        public long Id { get; set; }

        [Required]
        public long EmployeeId { get; set; }

        [ForeignKey(nameof(EmployeeId))]
        public virtual Employee Employee { get; set; } = default!;

        [Required]
        [MaxLength(100)]
        public string DocumentType { get; set; } = default!;

        [Required]
        [MaxLength(255)]
        public string OriginalFileName { get; set; } = default!;

        [Required]
        [MaxLength(500)]
        public string StorageKey { get; set; } = default!;

        [Required]
        [MaxLength(100)]
        public string MimeType { get; set; } = default!;

        [Required]
        public long SizeBytes { get; set; }

        [Required]
        public DateTime UploadedAtUtc { get; set; } = DateTime.UtcNow;
    }
}

File: neta-hiring-poc/python-api/app/schemas/employee_document.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class EmployeeDocumentCreate(BaseModel):
    employee_id: int
    document_type: str = Field(max_length=100)
    original_file_name: str = Field(max_length=255)
    storage_key: str = Field(max_length=500)
    mime_type: str = Field(max_length=100)
    size_bytes: int = Field(ge=0)


class EmployeeDocumentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    employee_id: int
    document_type: str
    original_file_name: str
    storage_key: str
    mime_type: str
    size_bytes: int
    uploaded_at_utc: datetime
